packages/itmat-interface/src/graphql/resolvers/userResolvers.ts

```diff
@@ -239,3 +336,45 @@ export const userResolvers = {
     },
     Subscription: {}
 };
+
+
+function formatEmailForForgottenPassword({ realname, username, to, resetPasswordToken }: { resetPasswordToken: string, to: string, username: string, realname: string }) {
+    const link = `${config.useSSL ? 'https' : 'http'}://${config.host}/resetPassword/${username}/${resetPasswordToken}`;
```

Comment:
This can be dangerous when using proxies or service the same application through multiple domains, which we are going to do here. I would maybe use the HTTP host as a base here

Reply:
i dont understand what you mean

Reply:
what HTTP host?

Reply:
This configuration implies that you are providing the domain name ahead of runtime. It can produce issues with SNI and can lead to problem with using a reverse proxy. If I serve the service on `ideafast.dsi.ic.ac.uk` but also on `data.ideafast.eu` I want the links to be generated based on what the user was visiting. There could also be firewall exceptions or package inspect mechanisms in place in a give network. Relying on the [HTTP request host](https://expressjs.com/en/5x/api.html#req.host) can avoid these issues